packages/flame/lib/src/components/core/component.dart

```diff
@@ -897,6 +901,8 @@ class Component {
 
 typedef ComponentSetFactory = ComponentSet Function();
 
+enum ChildrenChangeType { added, removed }
```

Comment:
Do we need this? 

Can't be just check if the given child if the child has the parent or not?

How do we envision this to work for reparenting?

Reply:
@spydon  suggested this additional type parameter for exactly not having to do this check.
Regarding reparenting: I'm not sure if i understand your question correctly. It also works for reparenting, whereas the old parent gets notified with a `removed` type, and the new parent with the `added` type.

Reply:
@spydon  suggested this additional type parameter for exactly not having to do this check.
Regarding reparenting: I'm not sure if i understand your question correctly. It also works for reparenting, whereas the old parent gets notified with a `removed` type, and the new parent with the `added` type (there is also a test for that case).

Reply:
This is a convenience design for the user to be able to easier determine whether a child has been added or removed. I don't think it would be clear to the user that they can check what the parent is to know whether it has been added or removed and this has no performance loss. 

> How do we envision this to work for reparenting?

It will follow the rest of the lifecycle flow that happens when reparenting.